lib/cib/cib_remote.c

```diff
@@ -384,28 +386,8 @@ cib_remote_signon(cib_t * cib, const char *name, enum cib_conn_type type)
     int rc = pcmk_ok;
     cib_remote_opaque_t *private = cib->variant_opaque;
 
-    if (private->passwd == NULL) {
-        struct termios settings;
-
-        rc = tcgetattr(0, &settings);
-        if(rc == 0) {
-            settings.c_lflag &= ~ECHO;
-            rc = tcsetattr(0, TCSANOW, &settings);
-        }
-
-        if(rc == 0) {
-            fprintf(stderr, "Password: ");
-            private->passwd = calloc(1, 1024);
-            rc = scanf("%1023s", private->passwd);
-            fprintf(stderr, "\n");
-        }
-
-        if (rc < 1) {
-            private->passwd = NULL;
-        }
-
-        settings.c_lflag |= ECHO;
-        rc = tcsetattr(0, TCSANOW, &settings);
+    if (private->passwd == NULL && private->out != NULL) {
+        private->out->prompt(private->out, "Password", false, &(private->passwd));
     }
```

Comment:
Hm, there's currently nothing about pcmk__text_prompt that actually uses the out argument (did I screw something up?), so we could just call it from here.  Should the out arg be important in the future, I could either create a formatted output object at this point or try to split pcmk__text_prompt apart into a part that uses out and a part that doesn't.

Reply:
Oh that's right, the prompt always goes to stderr, which makes sense.

Reply:
I should just get rid of the pcmk__output_t parameter and then I can call the function directly in cib_remote.c.

Reply:
Sure, but you either still need an output function so crm_mon curses can override it, or define a separate curses function and have crm_mon switch on the output format.

Reply:
Right, the function can still be a part of the pcmk__output_t function pointer table so the right format-specific one can be called.  It just doesn't need to take a pcmk__output_t * argument since it doesn't do anything with that.